extensions/exposition/source/HTTP/messages.ts

```diff
@@ -119,7 +117,9 @@ export interface OutgoingMessage {
   body?: any
 }
```

Comment:
Extending IncomingMessage by appending properties to it was the most criticised part of Express (and that's why Koa and Fastify implemented it differently) as it can easily result in memory leak and forever running request.

In modern days we should use [AsyncLocalStorage](https://nodejs.org/api/async_context.html#class-asynclocalstorage) for that.

Reply:
I agree that dynamic object extension is a bad idea in general. I'll refactor this into the wrapping type (`HTTPContext` or something).

AsyncLocalStorage is [unusable](https://github.com/DataDog/dd-trace-js/issues/1095) for production.

Reply:
I'm not sure that pointing to a 3 years old source is a good way to dismiss async context as I'm still can remember a lot of "benchmarks" from Node 4.x times clearly demonstrating that one should never use Promise "in production", and if you have to then you must bring the `blackbird` library with you (even MongoDb driver did that rolling back my PR to use async functions).

The beautiful part of standard library things is that they are improving with time rather than becoming a legacy.

Reply:
The fact that it is still open is more important than its age. There is no beauty significant enough to justify a 2x performance degradation (after [improvements](https://github.com/nodejs/node/issues/34493)).

Moreover, even if there were no performance impact at all, I can't see any reason in this specific task to choose ALS over passing the required object directly.

Reply:
You've just pointed to another 3 years old source (AsyncLocalStorage was under a flag at that time btw). But I would stop this discussion here - it seems that we are weighting the long term maintainability with two completely different factors, so it would be impossible to reach an agreement here. So, up to you for sure.

Reply:
We'll have to end the discussion when implementing telemetry.